pal/samples/esp8266/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.ino

```diff
@@ -0,0 +1,256 @@
+{
+    (void)reason;
+    (void)user_context;
+    // This sample DOES NOT take into consideration network outages.
+    if (result == IOTHUB_CLIENT_CONNECTION_AUTHENTICATED)
+    {
+        LogInfo("The device client is connected to iothub\r\n");
+    }
+    else
+    {
+        LogInfo("The device client has been disconnected\r\n");
+    }
+}
+
+void setup() {
+    int result = 0;
+
+    sample_init(ssid, pass);
+
+    // Create the iothub handle here
+    device_ll_handle = IoTHubDeviceClient_LL_CreateFromConnectionString(connectionString, protocol);
+    // Used to initialize IoTHub SDK subsystem
+    (void)IoTHub_Init();
+    
+    LogInfo("Creating IoTHub Device handle\r\n");
+    if (device_ll_handle == NULL)
+    {
+        LogInfo("Error AZ002: Failure createing Iothub device. Hint: Check you connection string.\r\n");
+    }
+    else
+    {
+        // Set any option that are neccessary.
+        // For available options please see the iothub_sdk_options.md documentation
+        // turn off diagnostic sampling
+        int diag_off = 0;
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_DIAGNOSTIC_SAMPLING_PERCENTAGE, &diag_off);
+
+#ifndef SAMPLE_HTTP
+        // Example sdk status tracing for troubleshooting
+        bool traceOn = true;
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_LOG_TRACE, &traceOn);
+#endif // SAMPLE_HTTP
+
+        // Setting the Trusted Certificate.
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_TRUSTED_CERT, certificates);
+
+#if defined SAMPLE_MQTT || defined SAMPLE_MQTT_WS
+        //Setting the auto URL Encoder (recommended for MQTT). Please use this option unless
+        //you are URL Encoding inputs yourself.
+        //ONLY valid for use with MQTT
+        bool urlEncodeOn = true;
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_AUTO_URL_ENCODE_DECODE, &urlEncodeOn);
+        /* Setting Message call back, so we can receive Commands. */
+        if (IoTHubClient_LL_SetMessageCallback(device_ll_handle, receive_message_callback, &receiveContext) != IOTHUB_CLIENT_OK)
+        {
+            LogInfo("ERROR: IoTHubClient_LL_SetMessageCallback..........FAILED!\r\n");
+        }
+#endif // SAMPLE_MQTT
+
+        // Setting connection status callback to get indication of connection to iothub
+        (void)IoTHubDeviceClient_LL_SetConnectionStatusCallback(device_ll_handle, connection_status_callback, NULL);
+
+        // action phase of the program, sending messages to the IoT Hub in the cloud.
+        do
+        {
+            if (messages_sent < MESSAGE_COUNT)
+            {
+                // Construct the iothub message from a string or a byte array
+                message_handle = IoTHubMessage_CreateFromString(telemetry_msg);
+                //message_handle = IoTHubMessage_CreateFromByteArray((const unsigned char*)msgText, strlen(msgText)));
```

Comment:
> //message_handle = IoTHubMessage_CreateFromByteArray((const unsigned char*)msgText, strlen(msgText))); [](start = 15, length = 103)

Why are we adding commented out code in this?

Reply:
In case people want to build a message from a ByteArray instead of a string? Do you think this option is a good one to have out there? I think it's in the C SDK's telemetry_sample as well.